pkg/orchestration/wiring/datadog/plugin.go

```diff
@@ -0,0 +1,201 @@
+package datadog
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	smith_v1 "github.com/atlassian/smith/pkg/apis/smith/v1"
+	"github.com/atlassian/voyager"
+	meta_orch "github.com/atlassian/voyager/pkg/apis/orchestration/meta"
+	orch_v1 "github.com/atlassian/voyager/pkg/apis/orchestration/v1"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringplugin"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/knownshapes"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/oap"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/svccatentangler"
+	sc_v1b1 "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/pkg/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	clusterServiceClassExternalID                      = "875d4a87-e887-4838-a0b5-b64491dbf9cb"
+	clusterServicePlanExternalID                       = "d8048a2d-49de-4fda-b7ef-328de171cd32"
+	ResourceType                  voyager.ResourceType = "datadog-alarm"
+)
+
+type WiringPlugin struct {
+	svccatentangler.SvcCatEntangler
```

Comment:
Sorry for a late suggestion, but we have recently introduced a cleaner way of using common code for OSB autowiring, and plan to migrate all plugins to it and get rid of SvcCatEntangler.
See https://github.com/atlassian/voyager/pull/73 and https://github.com/atlassian/voyager/pull/74/files as examples

If you prefer to finish the PR still using SvcCatEntangler that's fine, I will rewrite it later.